packages/angular/src/generators/web-worker/lib/normalize-options.ts

```diff
@@ -1,10 +1,37 @@
 import type { WebWorkerGeneratorOptions } from '../schema';
+import type { Tree } from '@nrwl/devkit';
+import { getProjects, joinPathFragments } from '@nrwl/devkit';
 
 export function normalizeOptions(
+  tree: Tree,
   options: WebWorkerGeneratorOptions
 ): WebWorkerGeneratorOptions {
+  if (!options.project) {
+    throw new Error('You must provide a project!');
+  }
+
+  const projects = getProjects(tree);
+  if (!projects.has(options.project)) {
+    throw new Error(`Project '${options.project}' does not exist.`);
+  }
+
+  const selectedProject = projects.get(options.project);
+  if (selectedProject.projectType !== 'application') {
+    throw new Error(
+      `Web Worker can only be added to an application. Project '${options.project}' is a library.`
+    );
+  }
+
+  const path =
+    options.path ?? joinPathFragments(selectedProject.sourceRoot, 'app');
```

Comment:
`sourceRoot` is optional, let's handle it:
```suggestion
  const path =
    options.path ?? selectedProject.sourceRoot
      ? joinPathFragments(selectedProject.sourceRoot, 'app')
      : joinPathFragments(selectedProject.root, 'src', 'app');
```